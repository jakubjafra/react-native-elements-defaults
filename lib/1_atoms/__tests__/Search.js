import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

const utils = require("../../utils");

describe('with no theme applied', () => {
    let OldSearch, Search;

    beforeAll(() => {
        let defaultStyles = {
        };

        OldSearch = require('react-native-elements/src/input/Search').default;
        Search = require('../Search').compose(defaultStyles)(OldSearch);
    });

    test('OldSearch and Search should be different', () => {
        expect(OldSearch).not.toBe(Search);
    });

    test('OldSearch renders correctly', () => {
        const tree = renderer.create(<OldSearch />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Search renders correctly', () => {
        const tree = renderer.create(<Search />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('OldSearch and Search returns the same thing if no theme is applied', () => {
        const oldTree = renderer.create(<OldSearch />).toJSON();
        const tree = renderer.create(<Search />).toJSON();

        expect(oldTree).toEqual(tree);
    });
});

describe('with basic theme applied', () => {
    let OldSearch, Search;

    beforeAll(() => {
        let defaultStyles = {
            style: {
                backgroundColor: 'red',
            }
        };

        OldSearch = require('react-native-elements/src/input/Search').default;
        Search = require('../Search').compose(defaultStyles)(OldSearch);
    });

    test('OldSearch and Search should be different', () => {
        expect(OldSearch).not.toBe(Search);
    });

    test('OldSearch renders correctly', () => {
        const tree = renderer.create(<OldSearch />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Search renders correctly', () => {
        const tree = renderer.create(<Search />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('OldSearch and Search should be different caused by theme applied', () => {
        const oldTree = renderer.create(<OldSearch />).toJSON();
        const tree = renderer.create(<Search />).toJSON();

        expect(JSON.stringify(oldTree)).not.toEqual(JSON.stringify(tree));
    });
});
