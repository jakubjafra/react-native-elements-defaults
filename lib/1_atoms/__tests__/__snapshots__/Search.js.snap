// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search with basic theme applied OldSearch renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#393e42",
        "borderBottomColor": "#000",
        "borderBottomWidth": 1,
        "borderTopColor": "#000",
        "borderTopWidth": 1,
      },
      false,
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    placeholderTextColor="#86939e"
    style={
      Array [
        Object {
          "backgroundColor": "#303337",
          "borderRadius": 3,
          "color": "#86939e",
          "fontSize": 17.5,
          "height": 30,
          "margin": 8,
          "overflow": "hidden",
          "paddingLeft": 26,
          "paddingRight": 19,
        },
        false,
        false,
        false,
        undefined,
        undefined,
        false,
      ]
    }
    underlineColorAndroid="transparent"
  />
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#86939e",
          "fontSize": 16,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
            "top": 15.5,
          },
          Object {
            "left": 16,
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î¢¶
  </Text>
</View>
`;

exports[`Search with basic theme applied Search renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#393e42",
        "borderBottomColor": "#000",
        "borderBottomWidth": 1,
        "borderTopColor": "#000",
        "borderTopWidth": 1,
      },
      false,
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    placeholderTextColor="#86939e"
    style={
      Array [
        Object {
          "backgroundColor": "#303337",
          "borderRadius": 3,
          "color": "#86939e",
          "fontSize": 17.5,
          "height": 30,
          "margin": 8,
          "overflow": "hidden",
          "paddingLeft": 26,
          "paddingRight": 19,
        },
        false,
        false,
        false,
        undefined,
        undefined,
        false,
      ]
    }
    underlineColorAndroid="transparent"
  />
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#86939e",
          "fontSize": 16,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
            "top": 15.5,
          },
          Object {
            "left": 16,
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î¢¶
  </Text>
</View>
`;

exports[`Search with no theme applied OldSearch and Search returns the same thing if no theme is applied 1`] = `"[2mCompared values have no visual difference.[22m"`;

exports[`Search with no theme applied OldSearch renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#393e42",
        "borderBottomColor": "#000",
        "borderBottomWidth": 1,
        "borderTopColor": "#000",
        "borderTopWidth": 1,
      },
      false,
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    placeholderTextColor="#86939e"
    style={
      Array [
        Object {
          "backgroundColor": "#303337",
          "borderRadius": 3,
          "color": "#86939e",
          "fontSize": 17.5,
          "height": 30,
          "margin": 8,
          "overflow": "hidden",
          "paddingLeft": 26,
          "paddingRight": 19,
        },
        false,
        false,
        false,
        undefined,
        undefined,
        false,
      ]
    }
    underlineColorAndroid="transparent"
  />
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#86939e",
          "fontSize": 16,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
            "top": 15.5,
          },
          Object {
            "left": 16,
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î¢¶
  </Text>
</View>
`;

exports[`Search with no theme applied Search renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#393e42",
        "borderBottomColor": "#000",
        "borderBottomWidth": 1,
        "borderTopColor": "#000",
        "borderTopWidth": 1,
      },
      false,
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    placeholderTextColor="#86939e"
    style={
      Array [
        Object {
          "backgroundColor": "#303337",
          "borderRadius": 3,
          "color": "#86939e",
          "fontSize": 17.5,
          "height": 30,
          "margin": 8,
          "overflow": "hidden",
          "paddingLeft": 26,
          "paddingRight": 19,
        },
        false,
        false,
        false,
        undefined,
        undefined,
        false,
      ]
    }
    underlineColorAndroid="transparent"
  />
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#86939e",
          "fontSize": 16,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
            "top": 15.5,
          },
          Object {
            "left": 16,
          },
          undefined,
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î¢¶
  </Text>
</View>
`;
